transcom/mymove
# Main b 18910 gun safe frontend

## [B-18910](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18910)

## INT PRs
#12458
#12512 

This branch adds the frontend changes necessary to allow an SC or TOO to indicate whether or not a customer is entitled to move a gun safe as part of their move.

1. There is now a checkbox on the "Edit Allowances" page with the label "Gun safe authorized" which allows the govt. user to indicate the gun safe status.
2. On the shipment card that is under the "Edit Allowances" link, a row has been added that indicates whether a gun safe has been authorized or not. This allows the user to easily see that their changes have been saved.

### How to test

1. Access the Move Details page of any move as an SC
2. Scroll down to the allowances card, verify that there is a new row at the bottom for gun safe, and that it is "Unauthorized".
3. Click the link, check the box for "Gun safe authorized", then click submit.
4. Scroll back down to the "Allowances" card and verify that the Gun Safe value is now "Authorized".
5. Repeat steps 1-4 while logged in as a TOO.

The new row in the shipment card:
![Screenshot 2024-04-10 at 10 07 23 AM](https://github.com/transcom/mymove/assets/146959895/9900cb47-21cd-4460-9f30-48a1a0d8755d)

The new checkbox on the Edit Allowances page:
![Screenshot 2024-04-10 at 10 08 06 AM](https://github.com/transcom/mymove/assets/146959895/bccc8407-8fa0-4f73-92b1-2ef0a19fd9fa)

Additionally, the gun safe entitlement is now visible to the Prime when they are viewing a move's information.

### How to test

1. Access the Move Details page of any move as an SC
2. Scroll down to the allowances card, verify that there is a new row at the bottom for gun safe, and that it is "Unauthorized".
3. Click the link, check the box for "Gun safe authorized", then click submit.
4. Scroll back down to the "Allowances" card and verify that the Gun Safe value is now "Authorized".
5. Repeat steps 1-4 while logged in as a TOO.
6. Now the move should be available to Prime, login as Prime and view the move, near the end of the move info, under "Diversion", there should now be a new field titled "Gun Safe", which should display the value you entered as SC/TOO.
